src/main/java/org/opentripplanner/gtfs/mapping/FareProductMapper.java

```diff
@@ -0,0 +1,69 @@
+package org.opentripplanner.gtfs.mapping;
+
+import java.time.Duration;
+import java.util.Currency;
+import org.opentripplanner.ext.fares.model.FareContainer;
+import org.opentripplanner.ext.fares.model.FareProduct;
+import org.opentripplanner.ext.fares.model.RiderCategory;
+import org.opentripplanner.routing.core.Money;
+
+public class FareProductMapper {
+
+  public static int NOT_SET = -999;
+
+  public FareProduct map(org.onebusaway.gtfs.model.FareProduct fareProduct) {
+    var id = AgencyAndIdMapper.mapAgencyAndId(fareProduct.getId());
+    var price = new Money(
+      Currency.getInstance(fareProduct.getCurrency()),
+      (int) (fareProduct.getAmount() * 100)
+    );
+
+    Duration duration = null;
+    if (fareProduct.getDurationUnit() != NOT_SET) {
+      duration = toTemporalUnit(fareProduct.getDurationUnit(), fareProduct.getDurationAmount());
+    }
+    return new FareProduct(
+      id,
+      fareProduct.getName(),
+      price,
+      duration,
+      mapRiderCategory(fareProduct.getRiderCategory()),
+      toInternalModel(fareProduct.egetFareContainer())
+    );
+  }
+
+  private static RiderCategory mapRiderCategory(
+    org.onebusaway.gtfs.model.RiderCategory riderCategory
+  ) {
+    if (riderCategory == null) {
+      return null;
+    } else {
+      return new RiderCategory(
+        riderCategory.getId().getId(),
+        riderCategory.getName(),
+        riderCategory.getEligibilityUrl()
+      );
+    }
+  }
+
+  private static Duration toTemporalUnit(int unit, int amount) {
+    return switch (unit) {
+      case 0 -> Duration.ofSeconds(amount);
+      case 1 -> Duration.ofMinutes(amount);
+      case 2 -> Duration.ofHours(amount);
+      case 3 -> Duration.ofDays(amount);
+      case 4 -> Duration.ofDays(amount * 7L);
+      case 5 -> Duration.ofDays(amount * 31L); // not totally right but good enough
+      case 6 -> Duration.ofDays(amount * 365L);
```

Comment:
We might want to use `java.time.Period` for these. Both implement `TemporalAmount`

Reply:
I tried that right now. However, I cannot convert a month to a number of seconds to check if it covers a journey.

```
java.time.temporal.UnsupportedTemporalTypeException: Unsupported unit: Seconds

	at java.base/java.time.Period.get(Period.java:443)
	at org.opentripplanner.ext.fares.model.FareProduct.coversDuration(FareProduct.java:32)
	at org.opentripplanner.gtfs.mapping.FareProductMapperTest.map(FareProductMapperTest.java:29)
```

Do you know how you can compare two TemporalAmounts?

Reply:
Periods and Durations don't mix so well as they cover slightly different concepts: one is about precise number of seconds the other one about a more complex notion of "a month".

Reply:
I think we need to extend this with `duration_type` in order to properly support the validity check. That way we can calculate the start of the validity and then use `java.time.temporal.TemporalAmount#addTo` to calculate the end of validity.

Reply:
We can add a todo here

Reply:
Done.